### PR TITLE
osd: don't update epoch and rollback_info if no need

### PR DESCRIPTION
**Pull request by mslovy**
there is no need to update epoch if no change, also rollback_info is just needed for ecpool, skip it when using replicated pool

Signed-off-by: Ning Yao <zay11022@gmail.com>

### CODE REVIEW

File: src/osd/PGLog.cc [hunk 1]
```diff
@@ -792,7 +794,8 @@ void PGLog::_write_log(
   const set<eversion_t> &trimmed,
   bool dirty_divergent_priors,
   bool touch_log,
-  set<string> *log_keys_debug
+  set<string> *log_keys_debug,
+  bool require_rollback
```
Comment (athanatos) [inline src/osd/PGLog.cc]: This should be before the optional pointer argument (per the style guide).

**Comment by athanatos** [comment]
This should be two commits.

I don't see where the require_rollback bool added through the write_log calls is actually used.

What sort of performance difference have you measured with this?

**Comment by mslovy** [comment]
@athanatos  fix and repush. In my test, it can reduce the execution time of omap_setkeys from 369us to 226us.  Actually, leveldb will loop WriteBatch and sort each key in Memtable based on SkipList. Less (key,value) pair will reduce the number of insertions, as well as LogEntry Preparation in leveldb.

File: src/osd/PGLog.cc (hunk 1 above)
Comment (mslovy) [inline src/osd/PGLog.cc]: fixed

**Comment by liewegas** [comment]
@mslovy can you break this into 2 commits please?

**Comment by mslovy** [comment]
@liewegas ,  is that ok?

**Comment by liewegas** [comment]
@mslovy also, small nit: in the commit messages, please prefix the first line with the subsystem.  for example, "osd: don't unnecessarily update epoch for pg"

**Comment by mslovy** [comment]
@liewegas, fix based on the comment, please review again

**Pull request merged**